### ISSUE TITLE
CMAKE Release build defaults to -O3 but RelWithDebInfo build defaults to -O2

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#507]
It looks like CMAKE Release build defaults to -O3 but RelWithDebInfo build defaults to -O2:

```
CMakeCache.txt
79:CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
103:CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
```

```
CMakeCache.txt
76:CMAKE_CXX_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
100:CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
```

When I'm working on Unvanquished/Daemon, I either do a Debug build to run on a debugger, or do a Release build to test for performance, meaning I never test the optimization level of our release builds,. On the other hand, that means I'm testing O3 since years without faciing specific issues.

There are maybe strong reason to use O2 when producing debug symbols, but on the other hand, maybe we would want to default to O3 in our released builds?

Note: doing O3 release builds may make Debian unhappy: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=640214

But I don't care that much as we're not integrated in their repo, and they are free to optimize less if they want to.

My main concern would be: how much O3 is problematic for debugging?

**Comment by slipher** [comment]
These default flag values come from CMake, not from us. It seems like people often question whether -O3 is actually better than -O2, so I don't see a strong reason to override it.

**Comment by illwieckz** [comment]
Yes those are CMake default, though maybe we may investigate the pros/cons to distribute `-O3` release builds. Most of the numbers from the [GPU compatibility matrix](https://wiki.unvanquished.net/wiki/GPU_compatibility_matrix) are then done with `-O3` because most of them were done with my pre 0.52 release local builds using `Release` CMake profile and then, `-O3`. I never use `RelWithDebInfo` CMake profile outside of release builds…
